DSharpPlus.Commands/Processors/TextCommands/TextExtensions.cs

```diff
@@ -0,0 +1,176 @@
+    public static DiscordMessage Copy(this DiscordMessage message) => messageConstructor(message);
+
+    /// <summary>
+    /// Creates a fake message created event args object with the specified content.
+    /// </summary>
+    /// <param name="context">The command context to use for the message.</param>
+    /// <param name="message">The message to use as a base for the new message.</param>
+    /// <param name="content">The content of the new message.</param>
+    /// <returns>The created message created event args object.</returns>
+    public static async ValueTask<MessageCreatedEventArgs> CreateFakeMessageEventArgsAsync(CommandContext context, DiscordMessage message, string content)
+    {
+        ArgumentNullException.ThrowIfNull(context, nameof(context));
+        ArgumentNullException.ThrowIfNull(message, nameof(message));
+        ArgumentException.ThrowIfNullOrWhiteSpace(content, nameof(content));
+
+        // Create a copy of the message to modify to prevent modifying the original message.
+        // By modifying the original message, we could potentially cause issues with the message cache.
+        DiscordMessage messageCopy = message.Copy();
+
+        // Modify the copied message to contain the contents that the user wants.
+        await messageCopy.ModifyMessagePropertiesAsync(content, context.Client, context.User, context.Channel, context.Guild);
+
+        // Create the message created event args.
+        MessageCreatedEventArgs messageCreateEventArgs = messageCreateEventArgsConstructor();
+        messageCreateEventArgsMessageSetter(messageCreateEventArgs, messageCopy);
+        messageCreateEventArgsMentionedUsersSetter(messageCreateEventArgs, messageCopy.MentionedUsers);
+        messageCreateEventArgsMentionedRolesSetter(messageCreateEventArgs, messageCopy.MentionedRoles);
+        messageCreateEventArgsMentionedChannelsSetter(messageCreateEventArgs, messageCopy.MentionedChannels);
+
+        // Return the message created event args, which can be used for argument conversion or command execution.
+        return messageCreateEventArgs;
+    }
+
+    /// <summary>
+    /// Modifies the internal properties of a message, changing the content, author, channel, and mentions.
+    /// </summary>
+    /// <param name="message">The message to modify.</param>
+    /// <param name="content">The new content of the message.</param>
+    /// <param name="client">The client to use for fetching users and channels.</param>
+    /// <param name="user">The new author of the message.</param>
+    /// <param name="channel">The new channel of the message.</param>
+    /// <param name="guild">The guild to use for fetching roles.</param>
+    public static async ValueTask ModifyMessagePropertiesAsync(
+        this DiscordMessage message,
+        string content,
+        DiscordClient client,
+        DiscordUser user,
+        DiscordChannel channel,
+        DiscordGuild? guild = null)

```

Comment:
inconsistent formatting with the codebase, the parentheses should be on entirely new lines